cli/src/cluster_query.rs

```diff
@@ -768,6 +772,32 @@ pub fn process_catchup(
             )
         };
 
+        let average_time_remaining = if total_sleep_interval == 0 {
+            "".to_string()
+        } else {
+            let distance_slot_variation = start_slot_distance as i64 - slot_distance as i64;
+            let average_catchup_slots_per_second = distance_slot_variation as f64 / f64::from(total_sleep_interval);
+            let average_time_remaining = (slot_distance as f64 / average_catchup_slots_per_second).round();
+            if !average_time_remaining.is_normal() {
+                "".to_string()
+            } else if average_time_remaining <= 0.0 {
+                format!(
+                    ". Average falling speed {:.1} slots/second",
+                    average_catchup_slots_per_second
+                )
+            } else {
+                let node_slot_variation = node_slot as i64 - start_node_slot as i64;
+                let average_node_slot_per_second = node_slot_variation as f64 / f64::from(total_sleep_interval);
+                let expected_finish_slot = (node_slot as f64 + average_time_remaining as f64 * average_node_slot_per_second as f64).round();
+                format!(
+                    ". Average catchup speed: {:.1} slots/second. Expected finish slot {} in {}",
```

Comment:
As you're concerned, code naming conversion doesn't necessarily need to be carried over to CLI output convention. Terminal space is quite precious resource. :)

` (AVG: {:.1} slots/second, ETA: slot {} in {})`

Reply:
Also, is there any reason you want to show the expected_finish_slot? how is it useful? I thought `average_remaining_time` is just fine.

Reply:
Yes, because I and other small validators try to update without skipping our leader slots. We find interval in schedule with size 1K-2K slots and start update in it. Expected slot will shows me that I'll be in time.

Reply:
> Yes, because I and other small validators try to update without skipping our leader slots.

Cool, the new `solana-validator wait-for-restart-window` command in v1.6 will do this for you automatically as well

Reply:
Oh, this is very good insight from validator operators like you. Admittedly, I couldn't notice these motives. Really appreciate for sharing your thinking. :)

How about commenting the insight like this?

```rust
// important not to miss next scheduled lead slots
let expected_finish_slot = ...
```